code/modules/mob/living/basic/minebots/minebot.dm

```diff
@@ -0,0 +1,191 @@
+		return ITEM_INTERACT_COMPLETE
+
+	return ..()
+
+/mob/living/basic/mining_drone/crowbar_act(mob/user, obj/item/I)
+	if(user.a_intent != INTENT_HELP)
+		return
+	. = TRUE
+	if(!I.tool_start_check(src, user, 0))
+		return
+	I.melee_attack_chain(user, stored_gun)
+
+/mob/living/basic/mining_drone/welder_act(mob/user, obj/item/I)
+	if(user.a_intent != INTENT_HELP)
+		return
+	. = TRUE
+	if(health == maxHealth)
+		to_chat(user, "<span class='notice'>[src] doesn't need repairing!</span>")
+		return
+	if(!I.tool_use_check(user, 1))
+		return
+	WELDER_ATTEMPT_REPAIR_MESSAGE
+	if(I.use_tool(src, user, 15, 1, volume = I.tool_volume) && health != maxHealth)
+		adjustBruteLoss(-20)
+		WELDER_REPAIR_SUCCESS_MESSAGE
+	return
+
+/mob/living/basic/mining_drone/death()
+	DropOre(0)
+	if(stored_gun)
+		for(var/obj/item/borg/upgrade/modkit/M in stored_gun.modkits)
+			M.uninstall(stored_gun)
+	robogibs(loc)
+	deathmessage = "blows apart!"
+	. = ..()
+
+/mob/living/basic/mining_drone/CanPass(atom/movable/O)
+	if(istype(O, /obj/item/projectile/kinetic))
+		var/obj/item/projectile/kinetic/K = O
+		if(K.kinetic_gun)
+			for(var/A in K.kinetic_gun.get_modkits())
+				var/obj/item/borg/upgrade/modkit/M = A
+				if(istype(M, /obj/item/borg/upgrade/modkit/minebot_passthrough))
+					return TRUE
+	if(istype(O, /obj/item/projectile/destabilizer))
+		return TRUE
+	return ..()
+
+/mob/living/basic/mining_drone/RangedAttack(atom/target, list/modifiers)
+	if(a_intent != INTENT_HARM)
+		return
+	stored_gun.afterattack__legacy__attackchain(target, src)
+
+/mob/living/basic/mining_drone/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+
+	if(!. || !proximity_flag || (a_intent == INTENT_HARM))
+		return
+
+	if(istype(attack_target, /obj/item/stack/ore))
+		var/obj/item/target_ore = attack_target
+		target_ore.forceMove(src)
+
+/mob/living/basic/mining_drone/proc/CollectOre()
+	for(var/obj/item/stack/ore/O in range(1, src))
+		O.forceMove(src)
+
+/mob/living/basic/mining_drone/proc/DropOre(message = 1)
+	if(!length(contents))
+		if(message)
+			to_chat(src, "<span class='warning'>You attempt to dump your stored ore, but you have none.</span>")
+		return
+	if(message)
+		to_chat(src, "<span class='notice'>You dump your stored ore.</span>")
+	for(var/obj/item/stack/ore/O in contents)
+		O.forceMove(drop_location())
+
+// MARK: Minebot Cube
+
+/obj/item/mining_drone_cube
+	name = "mining drone cube"
+	desc = "Compressed mining drone, ready for deployment. Just press the button to activate!"
+	w_class = WEIGHT_CLASS_SMALL
+	icon = 'icons/obj/aibots.dmi'
+	icon_state = "minedronecube"
+	item_state = "electronic"
+
+/obj/item/mining_drone_cube/attack_self__legacy__attackchain(mob/user)
+	user.visible_message("<span class='warning'>\The [src] suddenly expands into a fully functional mining drone!</span>", \
+	"<span class='warning'>You press center button on \the [src]. The device suddenly expands into a fully functional mining drone!</span>")
```

Comment:
The first one is cool since it capitalizes it
```suggestion
	"<span class='warning'>You press center button on [src]. The device suddenly expands into a fully functional mining drone!</span>")
```